api/consensus.go

```diff
@@ -19,6 +21,55 @@ type ConsensusGET struct {
 	Difficulty   types.Currency    `json:"difficulty"`
 }
 
+type ConsensusFileContract struct {
+	FileSize           uint64                         `json:"filesize"`
+	FileMerkleRoot     crypto.Hash                    `json:"filemerkleroot"`
+	WindowStart        types.BlockHeight              `json:"windowstart"`
+	WindowEnd          types.BlockHeight              `json:"windowend"`
+	Payout             types.Currency                 `json:"payout"`
+	ValidProofOutputs  map[string]types.SiacoinOutput `json:"validproofoutputs"`
+	MissedProofOutputs map[string]types.SiacoinOutput `json:"missedproofoutputs"`
+	UnlockHash         types.UnlockHash               `json:"unlockhash"`
+	RevisionNumber     uint64                         `json:"revisionnumber"`
+}
+
+type ConsensusFileContractRevision struct {
+	ParentID          types.FileContractID   `json:"parentid"`
+	UnlockConditions  types.UnlockConditions `json:"unlockconditions"`
+	NewRevisionNumber uint64                 `json:"newrevisionnumber"`
+
+	NewFileSize           uint64                         `json:"newfilesize"`
+	NewFileMerkleRoot     crypto.Hash                    `json:"newfilemerkleroot"`
+	NewWindowStart        types.BlockHeight              `json:"newwindowstart"`
+	NewWindowEnd          types.BlockHeight              `json:"newwindowend"`
+	NewValidProofOutputs  map[string]types.SiacoinOutput `json:"newvalidproofoutputs"`
+	NewMissedProofOutputs map[string]types.SiacoinOutput `json:"newmissedproofoutputs"`
+	NewUnlockHash         types.UnlockHash               `json:"newunlockhash"`
+}
+
+type ConsensusTransaction struct {
+	SiacoinInputs         map[string]types.SiacoinInput            `json:"siacoininputs"`
+	SiacoinOutputs        map[string]types.SiacoinOutput           `json:"siacoinoutputs"`
+	FileContracts         map[string]ConsensusFileContract         `json:"filecontracts"`
+	FileContractRevisions map[string]ConsensusFileContractRevision `json:"filecontractrevisions"`
```

Comment:
need this for child maps with hash as key
`ValidProofOutputs  map[string]types.SiacoinOutput`

but actually not sure that we need these output/input hashes, it should be in blockchain in usual way too as simple out/in not in contract?